ui/react-components/components/DatePicker/DatePicker.jsx

```diff
@@ -1,22 +1,42 @@
 import React, {Text, useState} from 'react';
 import Calendar from 'rc-calendar';
 import 'rc-calendar/assets/index.css';
-import { appointmentDatePicker, appointmentDatePickerSelected, appointmentDatePickerNotSelected } from './DatePicker.module.scss';
+import {
+    appointmentDatePicker,
+    appointmentDatePickerNotSelected,
+    appointmentDatePickerSelected
+} from './DatePicker.module.scss';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import moment from 'moment';
+import {FROM} from "../../constants";
 
 const AppointmentDatePicker = (props) => {
-    const [value, setValue] = useState(null);
+    const {isRecurring, startDate, startDateType, endDateType, dateType, defaultValue} = props;
+    const [value, setValue] = useState(defaultValue);
     const disablePastDates = (current) => {
         if (!current) {
             return false;
         }
         const date = moment().subtract('1', 'days').endOf('day');
-        return current.isBefore(date);
+        if (isRecurring) {
+            if (startDateType === FROM && dateType === "startDate")
+                return current.isBefore(date);
+            else if (endDateType === "On" && dateType === "endDate")
+                return current.isBefore(startDate);
+            else
+                return current;
+        } else
+            return current.isBefore(date);
+    };
```

Comment:
May be this logic should not be DatePicker

Reply:
yes changed